### ISSUE TITLE
how to train the model

### ISSUE THREAD

**Issue by Palashio** [kilgore92/PyTorch-UNet#1]
so i've created a pytorch model with this code: 
`model = UNet(n_channels=217,
             num_classes=2,
             use_bn=True,
             mode='3D',
             use_pooling=False
             )`

How do I actually train? My data is in two lists, original, and segmented (the first is the original and the second is the segmentation mask).

Additionally my image sizes are (217,512,512)

**Comment by Palashio** [comment]
For more information, each of my lists has 20 images of size (217, 512, 512)

**Comment by Palashio** [comment]
I tried model.forward() but i'm unsure of how to pass the masks and the original information to train.

**Comment by ishaanb92** [comment]
Hi Palash,

The repository only contains code for the model. It is intended to be used by installing the model as a package and importing it in your training script. For pointers on how to create a training script and a custom dataset class, I'd suggest looking at the tutorials/examples from the PyTorch website: https://pytorch.org/tutorials/

Cheers,
Ishaan

**Comment by ishaanb92** [comment]
Closing this issue, please open it if you need any further clarification.

**Comment by Palashio** [comment]
So, I've made pretty good progress:

```
for epoch in range(max_epochs):
    # Training
    for local_batch, local_labels in training_generator:
        # Transfer to GPU
        local_batch, local_labels = local_batch.to(device), local_labels.to(device)
```

this is where i'm at. basically `local_batch` has the original and `local_labels` has the segmented image in it. I built a custom DataLoader class to get it there. I'm a bit confused how to actually fit it to your UNet though. Could you provide any insights?

**Comment by Palashio** [comment]
nevermind fixed this issue.
